sdks/java/core/src/main/java/org/apache/beam/sdk/coders/StructuredCoder.java

```diff
@@ -98,6 +101,34 @@ public String toString() {
     return builder.toString();
   }
 
+  public void encode(T value, OutputStream outStream)
```

Comment:
Add default implementations of:
* encode(T, OutputStream, Context) that calls the appropriate encode(T, OutputStream)/encodeOuter(T, OutputStream) methods.
* decode(InputStream, Context) that calls the appropriate decode(InputStream)/decodeOuter(InputStream) methods.

This way we can remove the base method as soon as everyone has migrated to implementing the non Context version.

Also, we should have:
* encodeOuter(T, OutputStream) call encode(T, OutputStream)
* decodeOuter(InputStream) call decode(InputStream)

This way only Coders where outer context is important will need to implement it.